test/TestHelper.h

```diff
@@ -44,6 +44,41 @@ void connectClients(Client& c1, Client& c2);
 namespace test
 {
 
+/// Make sure that no Exception is thrown during testing. If one is thrown show its info and fail the test.
+/// Our version of BOOST_REQUIRE_NO_THROW()
+/// @param _expression    The expression for which to make sure no exceptions are thrown
+/// @param _message       A message to act as a prefix to the expression's error information
+#define ETH_TEST_REQUIRE_NO_THROW(_expression, _message)				\
+	do {																\
+		try															\
+		{																\
+			_expression;												\
+		}																\
+		catch (boost::exception const& _e)								\
+		{																\
+			auto msg = std::string(_message) + boost::diagnostic_information(_e); \
+			BOOST_FAIL(msg);											\
+		}																\
+	}while (0)
```

Comment:
This is the standard way to create *good* macros that can end in a semicolon and can act as an expression.